File: QzgfFrameNet/QzgfFrame.Mvc3/Content/Lib/js/Exam/ExamineeInfo.js
//var CurID =  <%=CurID %>;
var Validator = null;
var managerCompany;
$(function () {
    $.metadata.setType("attr", "validate");
    var v = $("form").validate({
        debug: true,
        errorPlacement: function (lable, element) {
            if (element.hasClass("dd-text")) element.ligerTip({ content: lable.html(), appendIdTo: lable });
            else if (element.hasClass("l-textarea")) element.ligerTip({ content: lable.html(), appendIdTo: lable });
            else if (element.hasClass("l-text-field")) element.parent().ligerTip({ content: lable.html(), appendIdTo: lable });
            else lable.appendTo(element.parents("td:first").next("td"));
        },
        success: function (lable) {
            lable.ligerHideTip();
        },
        submitHandler: function () {
            $("form .l-text,.l-textarea").ligerHideTip();
            f_save();
        }
    });
    $("form").ligerForm();
    deafultValidate($("form"));
    $("#ExamExamineeInfo_ExamTypeID").change(function () {
        var examtypestr = $("#ExamExamineeInfo_ExamTypeID").find("option:selected").text();
        selExamType(examtypestr);
    })
    selExamType($("#ExamExamineeInfo_ExamTypeID").val());
});

function selExamType(examtypestr) {
    if (examtypestr == "招聘") {
        $("#ExamExamineeInfo_FullName").removeAttr("readonly");
        $(":radio").attr("disabled", false);
        $("#ExamExamineeInfo_IDCardNumber").removeAttr("readonly");
        $("#ExamExamineeInfo_MobileNumber").removeAttr("readonly");
        $("#ExamExamineeInfo_DiplomaID").attr("disabled", false);
        $("#ExamExamineeInfo_BirthDate").attr("disabled", false);
        $("#selPersonnel").hide();
    }
    else {
        $("#ExamExamineeInfo_FullName").attr({ readonly: 'true' });
        $(":radio").attr("disabled", true);
        $("#ExamExamineeInfo_IDCardNumber").attr({ readonly: 'true' });
        $("#ExamExamineeInfo_MobileNumber").attr({ readonly: 'true' });
        $("#ExamExamineeInfo_DiplomaID").attr("disabled", true);
        $("#ExamExamineeInfo_BirthDate").attr("disabled", true);
        $("#selPersonnel").show();
    }
}

//选择人员信息
function f_selPersonnel() {
    var gridtype = "1";
    var URL = '/Exam/ExamineeInfo/SelectList/';
    var resu = window.showModalDialog(URL, window, 'help=0;status=0;dialogWidth=650px;dialogHeight=400px;');
    if (typeof (resu) != 'undefined') {
     for (var i = 0; i < resu.length; i++) {
            var stype = new Object();
            stype = resu[i];
            $("#ExamExamineeInfo_FullName").val(stype.FullName);
             if (stype.Sex == "0")
                 $("input[name='ExamExamineeInfo.Sex'][value=0]").attr("checked", true);
             else
                 $("input[name='ExamExamineeInfo.Sex'][value=1]").attr("checked", true);
             $("#ExamExamineeInfo_LoginName").val(stype.LoginName);
             $("#ExamExamineeInfo_IDCardNumber").val(stype.IDCardNumber);
             $("#ExamExamineeInfo_MobileNumber").val(stype.MobileNumber);
             $("#ExamExamineeInfo_DiplomaID").val(stype.DiplomaId);
             $("#ExamExamineeInfo_BirthDate").val(stype.BirthDate);
        }
    }
 }

 function f_checkData() {
     var LoginName = $("#ExamExamineeInfo_LoginName").val();
 $.ajax({
     type: 'GET', cache: false, dataType: 'json',
     url: "/exam/examineeinfo/Isloginnameexist?r='+new Date().getTime()",
     data: [
                   { name: 'Loginname', value: LoginName }
                   ],
     success: function (result) {
         if (result=="false") {
             $.ligerDialog.warn("考生登录名已存在!");
             return false;
         }
         else
             return true;
     },
     error: function (ex) {
         //debugger;
         //alert(ex);
         alert('发送系统错误,请与系统管理员联系!');
         return false;
     },
     beforeSend: function () {
     },
     complete: function () {
     }
 });

 }
 function f_save() {
     $(":radio").attr("disabled", false);        //取消不可用
     $("#ExamExamineeInfo_DiplomaID").attr("disabled", false);    //取消不可用
     $("#ExamExamineeInfo_BirthDate").attr("disabled", false);    //取消不可用
     $.ligerDialog.waitting("正在保存中...");
     //提交数据
     var fm = $("#newForm");
     //提交表单  result为返回的数据，success为成功,form为提交的form
     fm.ajaxSubmit(function (result, success, form) {
         if (result.Type) {
             $.ligerDialog.closeWaitting();
             if ($("#state").val() == "add") {
                 $.ligerDialog.confirm("是否继续添加记录?", '保存成功!', function (ok) {
                     if (ok) {
                         window.location.href = "/Exam/ExamineeInfo/add/0";
                     }
                     else
                         window.location.href = "/Exam/ExamineeInfo/";
                 });
             }
             else {
                 $.ligerDialog.success(result.Message, function () {
                     window.location.href = "/Exam/ExamineeInfo/";
                 });
             }
         }
         else {
             $.ligerDialog.closeWaitting();
             $.ligerDialog.error(result.Message, function () {
                 window.location.href = "/Exam/ExamineeInfo/";
             });
         }
     });
}

function f_windowclose() {
    window.location.href = "/Exam/ExamineeInfo/";
} 

File: QzgfFrameNet/QzgfFrame.Exam/ExamineeInfoManger/Models/ExamineeInfo.cs
/*
 * 文件名.........: ExamineeInfo.cs
 * 作者...........:  
 * 说明...........: 考生扩展信息类 
 * 注意...........: 
 * 修改记录.......:   时间       人员    备注
 *                    2011-01-15 XXXX 

*/

using System;
using QzgfFrame.Utility.Entity;

namespace QzgfFrame.Exam.ExamineeInfoManger.Models
{
	/// <summary>
	/// 考生信息表扩展
	/// </summary>
	[Serializable]
    public class ExamineeInfo
	{
        /// <summary>
        /// 考生信息
        /// </summary>
        public ExamExamineeInfo ExamExamineeInfo { get; set; }
        /// <summary>
        ///  考生ID
        /// </summary>
        public string ID { get; set; }
        /// <summary>
        ///  考生姓名
        /// </summary>
        public string FullName { get; set; }
        /// <summary>
        /// 性别
        /// </summary>
        public string Sex { get; set; }
        /// <summary>
        ///  考生登录名
        /// </summary>
        public string LoginName { get; set; }
        /// <summary>
        /// 身份证号码
        /// </summary>
        public string IDCardNumber { get; set; }
        /// <summary>
        /// 手机号码
        /// </summary>
        public string MobileNumber { get; set; }
        /// <summary>
        /// 学历文凭
        /// </summary>
        public string Diploma { get; set; }
        /// <summary>
        /// 学历文凭ID
        /// </summary>
        public string DiplomaID { get; set; }
        /// <summary>
        /// 出生日期
        /// </summary>
        public DateTime? BirthDate { get; set; }
        /// <summary>
        /// 考试类型
        /// </summary>
        public string ExamType { get; set; }
        /// <summary>
        /// 考生信息字符串
        /// </summary>
        public virtual string examineeinfos { get; set; }
	}
}